### DISCUSSION TITLE
Associations

### DISCUSSION THREAD

**Discussion by kyle-query** [ocsf/ocsf-schema#318] [Q&A]
What does the **associations** section at the bottom of some pages indicate? For instance, [Authentication](https://schema.ocsf.io/classes/authentication) has
* user: **dst_endpoint**
* src_endpoint: **actor.user**
* dst_endpoint: **user**
* actor.user: **src_endpoint**

[Authorization](https://schema.ocsf.io/classes/authorization) has
* user: **dst_endpoint**
* dst_endpoint: **user**

And all of the System Activity event classes, like [File Activity](https://schema.ocsf.io/classes/file_activity), have:
* device: **actor.user**
* actor.user **device**

I've searched this repo and ocsf-server for an explanation, but haven't found anything. I can make some guesses at what these mean in particular circumstances but I'm not really sure how to check if they are correct. Can anyone shed some light?

**Comment by rroupski** [comment]
@kyle-query sorry for the late response
Currently, the associations are between users and devices (endpoints). An event can have one or more users and one or more endpoints and it could be unclear where a user is defined.

Using the examples above, `actor.user` is on `src_endpoint`, and `user` is on `dst_endpoint `, in other words `actor.user` on `src_endpoint` _connects to_/_logs in on_ `dst_endpoint` as `user`.
